### PR TITLE
Add FairMQ/FairLogger as FairRoot deps for o2-fairroot-dev

### PR DESCRIPTION
**Pull request by dberzano**

### CODE REVIEW

**Comment by dberzano** [comment]
For the record, merging this PR will not change the dependencies in `defaults-o2` for the moment. Moreover, without FairRootGroup/FairRoot#778, it won't work anyways, even in `o2-dev-fairroot`. I propose to see if it does not break prod (i.e. green tests here) and merge it then. When the aforementioned FairRoot PR is merged (or the related problem is solved), let's make a new PR changing the prod behaviour too, and bumping FairRoot.

File: defaults-o2-dev-fairroot.sh [hunk 1]
```diff
@@ -61,6 +61,18 @@ overrides:
...
     source: https://github.com/FairRootGroup/FairRoot
+    requires:
+      - generators
+      - simulation
+      - ROOT
+      - ZeroMQ
+      - nanomsg
+      - boost
+      - protobuf
+      - DDS
```
Comment (dennisklein) [inline defaults-o2-dev-fairroot.sh]: ZeroMQ, nanomsg, DDS, and googletest can be dropped from FairRoot recipe. (Protobuf too, if you do not build the examples).

**Comment by dberzano** [comment]
@dennisklein so following your suggestion, FairRoot no longer depends on ZeroMQ, nanomsg and DDS. O2 already depended directly from DDS. Thanks for approving FairRootGroup/FairRoot#778!

**Comment by dberzano** [comment]
Since only o2-dev-fairroot is affected, I am merging this.

**Comment by awegrzyn** [comment]
@dberzano I got such error after merging this
````
[ 48%] Built target qcInfoServiceDump
[ 49%] Linking CXX executable ../bin/taskDPL
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld: warning: libppconsul.so, needed by /home/awegrzyn/alice/sw/slc7_x86-64/Configuration/v1.5.0-2/lib/libConfiguration.so, not found (try using -rpath or -rpath-link)
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld: warning: libjson11.so, needed by /home/awegrzyn/alice/sw/slc7_x86-64/Configuration/v1.5.0-2/lib/libConfiguration.so, not found (try using -rpath or -rpath-link)
/opt/rh/devtoolset-7/root/usr/libexec/gcc/x86_64-redhat-linux/7/ld: warning: libnanomsg.so.5.0.0, needed by /home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so, not found (try using -rpath or -rpath-link)
CMakeFiles/taskDPL.dir/src/TaskDPL.cxx.o: In function `main':
/home/awegrzyn/alice/sw/slc7_x86-64/O2/0c1bfb82d8_O2_DATAFLOW-1/include/Framework/runDataProcessing.h:98: undefined reference to `defineDataProcessing(o2::framework::ConfigContext const&)'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_close'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_getsockopt'
/home/awegrzyn/alice/sw/slc7_x86-64/Configuration/v1.5.0-2/lib/libConfiguration.so: undefined reference to `ppconsul::Consul::Consul(std::string const&, std::string const&, std::string const&)'
/home/awegrzyn/alice/sw/slc7_x86-64/Configuration/v1.5.0-2/lib/libConfiguration.so: undefined reference to `ppconsul::BadStatus::what() const'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_strerror'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_freemsg'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_errno'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_socket'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_reallocmsg'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_allocmsg'
/home/awegrzyn/alice/sw/slc7_x86-64/Configuration/v1.5.0-2/lib/libConfiguration.so: undefined reference to `typeinfo for ppconsul::BadStatus'
/home/awegrzyn/alice/sw/slc7_x86-64/Configuration/v1.5.0-2/lib/libConfiguration.so: undefined reference to `ppconsul::kv::impl::parseValues(std::string const&)'
/home/awegrzyn/alice/sw/slc7_x86-64/Configuration/v1.5.0-2/lib/libConfiguration.so: undefined reference to `ppconsul::helpers::encodeUrl(std::string const&)'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_recv'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_bind'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_connect'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_setsockopt'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_poll'
/home/awegrzyn/alice/sw/slc7_x86-64/Configuration/v1.5.0-2/lib/libConfiguration.so: undefined reference to `vtable for ppconsul::BadStatus'
/home/awegrzyn/alice/sw/slc7_x86-64/FairRoot/3b01d287cd-2/lib/libFairMQ.so: undefined reference to `nn_send'
collect2: error: ld returned 1 exit status
```

**Comment by dberzano** [comment]
Well, correlation does not imply causation :smile: The problem is surely unrelated if you did not use the `o2-dev-fairroot` default.

**Pull request merged**